### PR TITLE
layout onSwitch event for color scheme

### PR DESCRIPTION
**Pull request by mveres**

### CODE REVIEW

File: mdx-layout/src/StylesheetSwitch.tsx [hunk 1]
```diff
@@ -29,9 +29,11 @@ const labelStyle: CSSProperties = {
 export const StylesheetSwitch = ({
   defaultColorScheme = 'light',
   stylesheets,
+  onSwitch = () => {},
 }: {
   defaultColorScheme?: ColorScheme;
   stylesheets: Record<ColorScheme, string>;
+  onSwitch?: (scheme: 'dark' | 'light') => void;
```
Comment (gqio) [inline mdx-layout/src/StylesheetSwitch.tsx:36]: you can type with ColorScheme here I think

File: mdx-layout/src/index.tsx [hunk 2]
```diff
@@ -13,6 +13,7 @@ export const Layout = (props) => (
         light: 'https://cdn.jsdelivr.net/npm/water.css/out/light.min.css',
         dark: 'https://cdn.jsdelivr.net/npm/water.css/out/dark.min.css',
       }}
+      onSwitch={props.onSwitch}
```
Comment (gqio) [inline mdx-layout/src/index.tsx:16]: May I ask why do you need this event ? I used this technique with StyleSheetSwitch but maybe there is a better way to do this, I'm not an expert on React but I guess with useContext() we could achieve also some kind of useful propagation of event no ? I'm looking for guidance/advice

File: mdx-layout/src/StylesheetSwitch.tsx (hunk 1 above)
Comment (mveres) [reply mdx-layout/src/StylesheetSwitch.tsx:36]: fixed

File: mdx-layout/src/index.tsx (hunk 2 above)
Comment (mveres) [reply mdx-layout/src/index.tsx:16]: discussed this on discord

**Pull request merged**